ComputerVision/measurements.py

```diff
@@ -0,0 +1,170 @@
+        settings.imagelabels["measurementimage"].image = img
+        settings.images["measurementimage"] = img
+    else :
+        img = Image.fromarray(settings.images["imagetomeasure"])
+        img = ImageTk.PhotoImage(image = img)
+        settings.imagelabels["measurementimage"].configure(image=img)
+        settings.imagelabels["measurementimage"].image = img
+        settings.images["measurementimage"] = img
+
+def calculate_length (x1,x2,y1,y2) :
+    x,y,z = settings.images["imagetomeasure"].shape
+    inpixels = math.sqrt(math.pow(x1-x2,2)+math.pow(y1-y2,2))
+    distance = (settings.imagelength * inpixels) / x
+    settings.cellsizeidarray.append([settings.variables['cellid'], distance])
+
+def calculate () :
+    if len(settings.points) > 1 :
+        if settings.points[len(settings.points)-2][2] :
+            settings.points[len(settings.points)-2][2] = False
+            x1 = settings.points[len(settings.points)-1][0]
+            y1 = settings.points[len(settings.points)-1][1]
+            x2 = settings.points[len(settings.points)-2][0]
+            y2 = settings.points[len(settings.points)-2][1]
+            if y2 > y1 :
+                y3 = int(y2-((y2-y1)/2)-5)
+            else: 
+                y3 = int(y1-((y1-y2)/2)-5)
+            if x2 > x1 :
+                x3 = int(x2-((x2-x1)/2))
+            else: 
+                x3 = int(x1-((x1-x2)/2))
+            settings.previousImgs.append([settings.images["imagetomeasure"].copy(),"line"])
+            settings.images["imagetomeasure"] = cv2.line(settings.images["imagetomeasure"], (x1,y1), (x2,y2), (255, 0, 0), 2)
+            img = cv2.line(settings.previousImgsWithoutIDs[len(settings.previousImgsWithoutIDs)-1][0], (x1,y1), (x2,y2), (255, 0, 0), 2)
+            settings.previousImgsWithoutIDs.append([img])
+
+            settings.images["imagetomeasure"] = cv2.putText(settings.images["imagetomeasure"], str(settings.variables['cellid']), (x3, y3), cv2.FONT_HERSHEY_COMPLEX_SMALL, 2, (255, 0, 0), 2, cv2.LINE_AA)
+            calculate_length(x1,x2,y1,y2)  
+            settings.variables['cellid'] += 1
+            change_image()
+        else :
+            messagebox.showerror( "Error", "There is already a line here.")
+    else :
+       messagebox.showerror( "Error", "You need 2 points to create a calculation.")
+
+def add_measurement_buttons () :
+    settings.variables['cellid'] = 1
+    settings.frames['buttons'] = tk.Frame(master=settings.m, background="white")
+    settings.frames['buttons'].grid(row=0, column=3, sticky="n", rowspan=5)
+    settings.buttons['undoButton'] = tk.Button(master=settings.frames['buttons'], text="Undo Action", command=undo, width=30, height=2, padx=5, pady=5, background="white")
+    settings.buttons['undoButton'].pack()
+    settings.buttons['calculateButton'] = tk.Button(master=settings.frames['buttons'], text="Calculate", command=calculate, width=30, height=2, padx=5, pady=5, background="white")
```

Comment:
this could be extracted into a separate function that returns the button and accepts the button text as an argument

Reply:
Added the new function, works like a charm